### ISSUE TITLE
Promo banner is inaccessible to keyboard and screen reader users

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/vets-design-system-documentation#725]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## The problem

The [promo banner (VA Design System)](https://design.va.gov/storybook/?path=/docs/components-va-promo-banner--default) creates an inequivalent experience for screen reader, mobile, and keyboard users.

The promo banner's float property in particular can:

- hide keyboard focus for keyboard users
- cause clutter with other floating elements (like the feedback button and mobile menu)
- accidentally trigger unintended clicks and disorient users (especially since this is a full width touch target)
- create an unpredictable tabbing experience (keyboard users would be disorientated with a focus that jumps from the top, then to the bottom, and then back to the top)

It's also difficult for screen reader users to find as:
- it's not a landmark
- it doesn't have a heading
- it doesn't announce as an alert (and if it did, it would be very annoying)

https://user-images.githubusercontent.com/14154792/137928290-5c0acb34-ec61-4203-95e6-8a7fc083d70c.mov

Removing the float property, but styling it at the bottom of the page would still result in an unpredictable tabbing experience and it would no longer be visible on page load. 

Removing the float property and styling it at the top of the page would be better, but may result in an undesirably chunky header.

## Recommendation

I spent some time mocking 4 potential resolutions to the issues taking into account added complexity with:
- multiple alerts
- mobile
- multiple promo banners

![image](https://user-images.githubusercontent.com/14154792/164734729-4aa75886-48dd-4ebc-a4ba-0bfe5901bfdd.png)

### Promo banners and alerts have unique h2s above the h1 in a separate complementary landmark
- Pro: Used by NHS and Gov.UK. Uses complementary landmark with an aria-label which separates promo banners that are unique to the main content from the main
- Con: Places h2 above the h1 which may confuse some folks who aren't familiar with the pattern. Also adds volume to the landmark list. 

### A status region contains all promo banners and alerts as a list without any headings inbetween the header and the main
- Pro: Announces to all users upon loading on the page in a polite fashion.
- Con: Not discoverable by headings. Needs review to see if multiple status announcements (aria-polite) at once can announce poorly or create issues

### Promo banner and alerts are nested in the header 
- Pros: Works well for promo banners like site maintenance as it's being shared on all pages at all times. Place it early in the DOM makes it more easy to discover.
- Con: Not discoverable by headings and can bloat the header.

### Promo banners and alerts are a part of the main as h2s after the h1
- Pro: h2s are ordered below the h1 making announcements easily discoverable and logically ordered
- Con: Does not split unrelated content like site maintenance into a separate landmark.

If we decide to jump into this discussion more, I can create codepen prototypes featuring each of these. Would also love to invite feedback on this from others too.

## Related tickets and conversations
[Conversation with Dave, Ryan, and Kelson](https://dsva.slack.com/archives/C52CL1PKQ/p1635952698280200), this proposal was tabled by Dave for the following reasons:
- Let's worry about getting the content managed by the CMS.
- Hold off on the other fixes.
- Here's why: We're (Ryan) looking at redesiging the homepage in totality over the next few months and I'd like to consider both 508-first alongside Mobile first as we rebuild, rather than piecemeal this and possibly have throwaway work.

Several other va-alert tickets address padding issues on mobile. I hope these designs can help further that conversation 🙇

**Comment by humancompanion-usds** [comment]
@joshkimux - Which 508 defect level would you assign to this? I want to label it appropriately and want to get your input on the severity. By your description I would say a 1 or a 2. Does that seem right?

**Comment by joshkimux** [comment]
[I assigned this originally as a defect-level-2](https://github.com/department-of-veterans-affairs/va.gov-team/issues/32285) ✌️

**Comment by bkjohnson** [comment]
@joshkimux regarding this acceptance criteria:

> The promo banner's click target is limited to the link itself, not the full component

I believe that it is [already satisfied](https://github.com/department-of-veterans-affairs/component-library/blob/3fc3b7d33d9cfe5ac51a54d7aed0b445c26cfe05/packages/web-components/src/components/va-promo-banner/va-promo-banner.tsx#L129-L133), the actual link just takes up a lot of space:

![Screenshot of a promo banner component on Storybook with the inner `<a>` highlighted revealing that it expands to the right to occupy empty space](https://user-images.githubusercontent.com/2008881/184425153-a7ef2662-9bd2-44b6-ba14-c2ee33ab6488.png)

**Comment by joshkimux** [comment]
@bkjohnson thanks for clarifying, I'll update the acceptance criteria! Ideally we'd like the link to not be full width beyond the text itself to prevent misclicks on mobile screen readers (because this is sticky and full width).

**Comment by joshkimux** [comment]
Updated this to include a screenshot from @cgednalske on how big the promo banner can be at 175% zoom for Veterans with low vision 
![image](https://user-images.githubusercontent.com/14154792/193096118-7f1edc5a-4333-4fc3-984f-52345242c4f4.png)

**Comment by TiffanyPender** [comment]
I recently came across a related promo banner issue when doing an accessibility audit of the newly redesigned VA.gov home page.  When doing screen reader testing, there are two landmarks that appear in the rotor that are not actually visible on the page. This means that although nothing is visible, to assistive technology users they may mistakenly think something is there. Both have a role=“region”. One has a data-widget-type=“maintenance-banner”, and the other is a <va-promo-banner> related to the PACT Act.
Additional details can be found in [GitHub issue #11827](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11827) (closed as it was out-of-scope for the homepage redesign). It seems that the empty maintenance banner is an existing backlog issue that occurs site-wide.

**Comment by humancompanion-usds** [comment]
@wesrowe - Is Promo banner no longer with the home page update? Or does it still lurk on va.gov?

**Comment by wesrowe** [comment]
I don't see any reason for it not to be available. Did you expect it somewhere it didn't appear?

**Comment by caw310** [comment]
Closing!
